apps/shell/app.js

```diff
@@ -52,18 +54,38 @@ app.set('views', path.join(__dirname, 'views'));
 app.use(process.env.PASSENGER_BASE_URI || '/', router);
 
 // Setup websocket server
-var server = new http.createServer(app);
-var wss = new WebSocket.Server({ noServer: true });
+const server = new http.createServer(app);
+const wss = new WebSocket.Server({ noServer: true });
 
-wss.on('connection', function connection (ws, req) {
-  var match;
-  var host = process.env.DEFAULT_SSHHOST || 'localhost';
-  var cmd = process.env.OOD_SSH_WRAPPER || 'ssh';
-  var dir;
-  var term;
-  var args;
-  var host_path_rx = '/ssh/([^\\/\\?]+)([^\\?]+)?(\\?.*)?$';
+let whitelist = [];
+if (process.env.SSHHOST_WHITELIST){
+  whitelist = process.env.SSHHOST_WHITELIST.split(':');
+}
+
+glob.sync('/etc/ood/config/clusters.d/*.y*ml')
+  .map(yml => yaml.safeLoad(fs.readFileSync(yml)))
+  .filter(config => (config.v2 && config.v2.login && config.v2.login.host) && ! (config.v2 && config.v2.metadata && config.v2.metadata.hidden))
+  .forEach((config) => {
+    let host = config.v2.login.host; //Already did checking above
+    let isDefault = config.v2.login.default;
+    if(isDefault){
+      whitelist.unshift(host);
+    } else{
+      whitelist.push(host);
+    }
+  });
+
+whitelist = Array.from(new Set(whitelist)); // remove duplicates
+const default_sshhost = whitelist[0]; //default sshhost if not set
 
+wss.on('connection', function connection (ws, req) {
+  var match,
+      dir,
+      term,
+      args,
+      host = process.env.DEFAULT_SSHHOST || default_sshhost,
+      cmd = process.env.OOD_SSH_WRAPPER || 'ssh',
+      host_path_rx = '/ssh/([^\\/\\?]+)([^\\?]+)?(\\?.*)?$';
   console.log('Connection established');
```

Comment:
We should establish a consistent way to log around the application. I recommend we take the following approach:

When defining our variables at the start of this file (line 14 is a good place), redirect log to console.log
```js
const log = console.log
```

```suggestion
  log('Connection established');
```

And then throughout the application replace `console.log('Some logs')` with `log('Some logs')`. The advantage to doing it this way is that we're creating a "middleware" for all logs throughout the application so in the future if we wanted to modify the log format we could modify `const log = console.log` to: 
```js
const log = function (in) {
   // do something
}
```
Doing it this way means we won't have to go throughout the application and remember where we have `console.log()` we just modify the log "middleware"